### PR TITLE
[ENH] Test coverage for MLP Network improved

### PR DESCRIPTION
**Pull request by shinymack**
#### Reference Issues/PRs
Contributes to #2497 

#### What does this implement/fix? Explain your changes.
Adds test coverage for _mlp.py

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### Any other comments?
Achieves 96% coverage on _mlp.py, checked using 
```pytest --cov=aeon.networks aeon/networks/ ```

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you after the PR has been merged.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

File: aeon/networks/tests/test_mlp.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        (5, [256, 128, 128, 64, 32], "sigmoid"),
+        (2, 128, ["tanh", "relu"]),
+    ],
+)
+def test_mlp_initialization(n_layers, n_units, activation):
+    """Test whether MLPNetwork initializes correctly with different configurations."""
+    mlp = MLPNetwork(n_layers=n_layers, n_units=n_units, activation=activation)
+    input_layer, output_layer = mlp.build_network((1000, 5))
+    assert input_layer is not None
+    assert output_layer is not None
```
Comment (baraline) [inline aeon/networks/tests/test_mlp.py:26]: Same comment as in another PR, the tests should explicitly test for the expected object type rather than none.
Additionally, testing that those objects have the expected number of layers/units/the correct activation function would be nice to really cover the estimator.

**Review by baraline — CHANGES_REQUESTED** [review]
The first comment applies for all tests.

File: aeon/networks/tests/test_mlp.py (hunk 1 above)
Comment (shinymack) [reply aeon/networks/tests/test_mlp.py]: Thanks for the feedback! I've updated the tests to explicitly check for the expected object type instead of just None. Additionally, I’ve added assertions to verify the number of layers, units, and activation functions where applicable to ensure better coverage of the estimator. Let me know if you have any further suggestions!

**Comment by shinymack** [comment]
@baraline @hadifawaz1999, I've updated the tests to explicitly assert the expected layer types and verify the model parameters. Initially, isinstance(input_layer, InputLayer) wasn't working because input_layer is a KerasTensor, not a direct layer instance. To resolve this, I accessed _keras_history[0], which gives the actual layer. Let me know if this approach aligns with what you were expecting.

Also, I ensured TensorFlow is treated as a soft dependency by moving imports inside the test functions, so the CI should now pass without issues. Please review the changes and let me know if any further modifications are needed.

Thanks for your time!

**Comment by baraline** [comment]
Thanks for the changes, I'm not very familiar with TensorFlow, so I'll let Ali confirm about the use of `_keras_history[0]`, but about the parameters, I was thinking more like what you did here `output_layer_actual.rate == dropout_last`.

The goal is to test that we've built a the network that matches the parameter that we define in `__init__` (i.e that we don't forget to give those parameters when building the network), hence testing directly from the estimator (e.g. `mlp.dropout_last == dropout_last`) is not what I had in mind, but more like what you did for `output_layer_actual.rate == dropout_last`.

**Comment by hadifawaz1999** [comment]
> Thanks for the changes, I'm not very familiar with TensorFlow, so I'll let Ali confirm about the use of `_keras_history[0]`, but about the parameters, I was thinking more like what you did here `output_layer_actual.rate == dropout_last`.
> 
> The goal is to test that we've built a the network that matches the parameter that we define in `__init__` (i.e that we don't forget to give those parameters when building the network), hence testing directly from the estimator (e.g. `mlp.dropout_last == dropout_last`) is not what I had in mind, but more like what you did for `output_layer_actual.rate == dropout_last`.

I would prefer not using ```_keras_history``` as it is unknown if it will still exists in new versions to come, so i would instead do as @baraline is proposing and check if the internal parameters of the layers are correct, and if they are none or not, also their output shpaes

**Comment by shinymack** [comment]
@baraline @hadifawaz1999 Thanks for the feedback! I’ve updated the tests to avoid using _keras_history and instead wrapped the returned input_layer and output_layer in a Model. This way, I can directly inspect model.layers to verify the structure and parameters, making the checks more reliable and future-proof. Let me know if this looks good or if there’s anything else you’d suggest!

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thanks for taking care of this, but could you explicity say "mlp" in the test function names ? this way we are sure of the test association

**Comment by shinymack** [comment]
> Thanks for taking care of this, but could you explicity say "mlp" in the test function names ? this way we are sure of the test association

Updated the test function names to include 'mlp'. Let me know if anything else is needed!

**Review by baraline — APPROVED** [review]
LGTM, thanks ! 
Small note, next time, try to avoid using force pushing, it prevents us from seeing the changes made between each commit correctly when reviewing.

**Review by hadifawaz1999 — APPROVED** [review]
LGTM thanks

**Comment by hadifawaz1999** [comment]
> LGTM, thanks ! Small note, next time, try to avoid using force pushing, it prevents us from seeing the changes made between each commit correctly when reviewing.

I agree

**Comment by shinymack** [comment]
Got it! I'll keep that in mind and avoid force pushing from next time. Thanks for your time and feedback!

**Comment by shinymack** [comment]
@all-contributors please add @shinymack for code
